cscs-checks/microbenchmarks/gpu/memory_bandwidth/src/memory_bandwidth.cu

```diff
@@ -0,0 +1,77 @@
+#include "testHeaders.hpp"
+
+int main()
+{
+
+  char * nid_name = (char *)calloc(80,sizeof(char));
+  gethostname(nid_name,80);
+
+  int number_of_devices;
+  XGetDeviceCount(number_of_devices);
+
+  // Initialise the Smi to manage the devices.
+  Smi smiHandle;
+
+  // Make sure we've got devices aboard.
+  if (number_of_devices == 0)
+  {
+    std::cout << "No devices found on host " << nid_name << std::endl;
+    return 1;
+  }
+  else
+  {
+    printf("[%s] Found %d device(s).\n", nid_name, number_of_devices);
+  }
+
+  /*
+   Test parameters
+  */
+  size_t copy_size = COPY_SIZE;
+  int copy_repeats = NUMBER_OF_COPIES;
+  float fact = (float)copy_size/(float)1e3;
+
+  /*
+   Test the Host to Device bandwidth.
+  */
+  for (int d = 0; d < number_of_devices; d++)
+  {
+    XSetDevice(d);
+    smiHandle.setCpuAffinity(d);
+    float bw = fact / copyBandwidth<HostData, DeviceData>(copy_size, d, copy_repeats, XMemcpyHostToDevice);
+    printf("[%s] Host to device bandwidth on device %d is %.2f Mb/s.\n", nid_name, d, bw);
```

Comment:
Why not `std::cout` here? (NB: I don't have a problem, just wondering).

Reply:
It's shorter :)

Reply:
Haha, I was expecting that answer :-D